test/EPubTests.ts

```diff
@@ -0,0 +1,452 @@
+              { "@attributes": { idref: "id_cover_xhtml", linear: "yes" } },
+              { "@attributes": { idref: "titlepage", linear: "yes" } },
+              { "@attributes": { idref: "copyright", linear: "yes" } },
+              { "@attributes": { idref: "toc", linear: "yes" } },
+              { "@attributes": { idref: "dedication", linear: "yes" } },
+              { "@attributes": { idref: "chapter001", linear: "yes" } },
+              { "@attributes": { idref: "chapter002", linear: "yes" } },
+            ],
+          },
+          guide: {
+            reference: [
+              {
+                "@attributes": {
+                  type: "copyright",
+                  title: "Copyright",
+                  href: "copyright.xhtml",
+                },
+              },
+              {
+                "@attributes": {
+                  type: "start",
+                  title: "Begin Reading",
+                  href: "cover.xhtml",
+                },
+              },
+              {
+                "@attributes": {
+                  type: "toc",
+                  title: "Table of Contents",
+                  href: "toc.xhtml",
+                },
+              },
+              {
+                "@attributes": {
+                  type: "cover",
+                  title: "Cover Image",
+                  href: "cover.xhtml",
+                },
+              },
+            ],
+          },
+        },
+      },
+      new URL("http://example.com/manifest.json")
+    );
+
+    emptyManifest = new Manifest(
+      {},
+      new URL("http://example.com/manifest.json")
+    );
+  });
+
+  describe("#getManifest", () => {
+    const manifestJSON = {
+      metadata: {
+        "dc:title": "The Elephant",
+      },
+    };
+    const manifest = new Manifest(
+      manifestJSON,
+      new URL("https://example.com/manifest.json")
+    );
+    let store: MemoryStore;
+
+    beforeEach(() => {
+      store = new MemoryStore();
+    });
+
+    describe("if fetching the manifest fails", () => {
+      beforeEach(() => {
+        window.fetch = stub().rejects();
+      });
+
+      it("should return cached manifest from local store", async () => {
+        const key = "manifest";
+        await store.set(key, JSON.stringify(manifestJSON));
+
+        const response: Manifest = await Manifest.getManifest(
+          new URL("https://example.com/manifest.json"),
+          store
+        );
+        expect(response).to.deep.equal(manifest);
+      });
+
+      it("should reject promise if there's nothing in the store", async () => {
+        let rejected = false;
+        try {
+          await Manifest.getManifest(
+            new URL("https://example.com/manifest.json"),
+            store
+          );
+        } catch (err) {
+          rejected = true;
+        }
+        expect(rejected).to.be.true;
+      });
+    });
+
+    it.skip("should return the response from fetch, and save it to local store", async () => {
```

Comment:
I modified this test suite from Manifest.ts but need to understand it better. I am not sure why xx is returning the manifest but saying that an empty response was expected 🤔 

  ```AssertionError: expected { Object (metadata, links, ...) } to deeply equal { Object (metadata, links, ...) }
      + expected - actual

       {
         "links": []
         "manifestUrl": {}
      -  "metadata": {
      -    "title": "The Elephant"
      -  }
      +  "metadata": {}
         "resources": []
         "spine": []
         "toc": []
       }``` I am not sure why it's expecting an empty object, I am expecting ..

If I switch expected to `{
          links: [],
          manifestUrl: {},
          metadata: {
            title: "The Elephant",
          },
          resources: [],
          spine: [],
          toc: [],
        }` it fails with `AssertionError: expected { Object (metadata, links, ...) } to deeply equal {}`